drivers/serial/uart_nrfx_uart.c

```diff
@@ -413,7 +413,7 @@ DEVICE_AND_API_INIT(uart_nrfx_uart0,
 		    uart_nrfx_init,
 		    NULL,
 		    NULL,
-		    POST_KERNEL,
+		    PRE_KERNEL_1,
```

Comment:
What about adding some comment here explaining why it has to be initialized at PRE_KERNEL_1, so that such unwanted change is not done again in some future?

Reply:
Done, there is also a similar comment in uart_console.c explaining the init level and priority.